src/modules/ekf2/EKF/aid_sources/aux_global_position/aux_global_position.cpp

```diff
@@ -42,147 +42,159 @@ void AuxGlobalPosition::update(Ekf &ekf, const estimator::imuSample &imu_delayed
+			case State::kStarting:
+				if (starting_conditions) {
+					source.state = State::kStarting;
+
+					if (ekf.global_origin_valid()) {
+						const bool fused = ekf.fuseHorizontalPosition(aid_src);
+						bool reset = false;
+
+						if (!fused && isResetAllowed(ekf, slot)) {
+							ekf.resetHorizontalPositionTo(sample.latitude, sample.longitude, Vector2f(aid_src.observation_variance));
+							ekf.resetAidSourceStatusZeroInnovation(aid_src);
+							reset = true;
+						}
+
+						if (fused || reset) {
+							ekf.enableControlStatusAuxGpos();
+							source.reset_counters.lat_lon = sample.lat_lon_reset_counter;
+							source.state = State::kActive;
+						}
 
-				} else {
-					// Try to initialize using measurement
-					if (ekf.resetGlobalPositionTo(sample.latitude, sample.longitude, sample.altitude_amsl, pos_var,
-								      sq(sample.epv))) {
-						ekf.resetAidSourceStatusZeroInnovation(aid_src);
-						ekf.enableControlStatusAuxGpos();
-						_reset_counters.lat_lon = sample.lat_lon_reset_counter;
-						_state = State::kActive;
+					} else {
+						// Try to initialize using measurement
+						if (ekf.resetGlobalPositionTo(sample.latitude, sample.longitude, sample.altitude_amsl, pos_var,
+									      sq(sample.epv))) {
+							ekf.resetAidSourceStatusZeroInnovation(aid_src);
+							ekf.enableControlStatusAuxGpos();
+							source.reset_counters.lat_lon = sample.lat_lon_reset_counter;
+							source.state = State::kActive;
+						}
 					}
 				}
-			}
 
-			break;
+				break;
 
-		case State::kActive:
-			if (continuing_conditions) {
-				ekf.fuseHorizontalPosition(aid_src);
+			case State::kActive:
+				if (continuing_conditions) {
+					ekf.fuseHorizontalPosition(aid_src);
 
-				if (isTimedOut(aid_src.time_last_fuse, imu_delayed.time_us, ekf._params.reset_timeout_max)
-				    || (_reset_counters.lat_lon != sample.lat_lon_reset_counter)) {
-					if (isResetAllowed(ekf)) {
+					if (isTimedOut(aid_src.time_last_fuse, imu_delayed.time_us, ekf._params.reset_timeout_max)
+					    || (source.reset_counters.lat_lon != sample.lat_lon_reset_counter)) {
+						if (isResetAllowed(ekf, slot)) {
 
-						ekf.resetHorizontalPositionTo(sample.latitude, sample.longitude, Vector2f(aid_src.observation_variance));
+							ekf.resetHorizontalPositionTo(sample.latitude, sample.longitude, Vector2f(aid_src.observation_variance));
 
-						ekf.resetAidSourceStatusZeroInnovation(aid_src);
+							ekf.resetAidSourceStatusZeroInnovation(aid_src);
 
-						_reset_counters.lat_lon = sample.lat_lon_reset_counter;
+							source.reset_counters.lat_lon = sample.lat_lon_reset_counter;
 
-					} else {
-						ekf.disableControlStatusAuxGpos();
-						_state = State::kStopped;
+						} else {
+							ekf.disableControlStatusAuxGpos();
+							source.state = State::kStopped;
+						}
 					}
-				}
 
-			} else {
-				ekf.disableControlStatusAuxGpos();
-				_state = State::kStopped;
-			}
+				} else {
+					ekf.disableControlStatusAuxGpos();
```

Comment:
shouldn't this (`_control_status.flags.aux_gpos`) be slot-specific too?
one slot could fail and stop aiding but the others could still be healthy.